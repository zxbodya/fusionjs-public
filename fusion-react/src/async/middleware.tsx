/** Copyright (c) 2018 Uber Technologies, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */

import * as React from 'react';

import type {Middleware} from 'fusion-core';

import PrepareProvider from './prepare-provider';

const middleware: Middleware = function (ctx, next) {
  if (__DEV__) {
    // eslint-disable-next-line no-console
    console.warn('The {middleware} export from fusion-react is deprecated.');
  }

  if (__NODE__ && !ctx.element) {
    return next();
  }

  const markAsCritical = __NODE__
    ? (chunkId) => {
        // Push to legacy context for backwards compat w/ legacy SSR template
        ctx.preloadChunks.push(chunkId);
      }
    : noop;
  ctx.element = (
    <PrepareProvider markAsCritical={markAsCritical}>
      {ctx.element}
    </PrepareProvider>
  );
  return next();
};

export default middleware;

function noop() {}
