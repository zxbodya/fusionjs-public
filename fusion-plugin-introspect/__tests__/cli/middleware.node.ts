// @flow
import {mockFs} from './mockFs.js';
import {middleware} from '../../src/cli/middleware.js';

test('middleware', async () => {
  mockFs({
    version: '0.0.0',
    server: [
      {
        timestamp: 0,
        dependencies: [
          {
            name: 'A',
            type: 'both',
            sources: [{type: 'plugin', source: 'foo/bar:1:2'}],
            dependencies: ['B'],
          },
          {
            name: 'B',
            type: 'both',
            sources: [{type: 'plugin', source: 'foo/bar:3:4'}],
            dependencies: ['C'],
          },
          {
            name: 'C',
            type: 'both',
            sources: [{type: 'plugin', source: 'foo/bar:5:6'}],
            dependencies: [],
          },
        ],
      },
    ],
  });
  const data = await middleware();
  expect(data).toMatchSnapshot();
});
