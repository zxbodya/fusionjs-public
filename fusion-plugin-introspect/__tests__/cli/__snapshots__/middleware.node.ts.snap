// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middleware 1`] = `
"Server:
C  foo/bar:5:6
B  foo/bar:3:4
A  foo/bar:1:2

"
`;
